plutus-core/src/Language/PlutusCore/Constant/Typed.hs

```diff
@@ -340,3 +351,36 @@ instance KnownBuiltinType term String        => KnownType term String
 instance KnownBuiltinType term Char          => KnownType term Char
 instance KnownBuiltinType term ()            => KnownType term ()
 instance KnownBuiltinType term Bool          => KnownType term Bool
+
+{- Note [Int as Integer]
+We represent 'Int' as 'Integer' in PLC and check that an 'Integer' fits into 'Int' when
+unlifting constants (and fail if it doesn't).
+
+There's a catch, though: an out-of-bounds error is not an internal one -- it's a normal
+evaluation failure, but unlifting errors have this connotation of being "internal".
+We could make 'readKnown' return an @m (EvaluationResult a)@, but that makes it more awkward
+to use (in particular, in functions like 'readKnownCek').
+
+Another option is be to define and use @AsEvaluationFailure@ in addition to 'AsUnliftingError'.
```

Comment:
You did this in the end, right? so the comment is out of date

Reply:
Yes, the docs are not updated (I did not want to update them before you approve the solution). Will do.